Jenkinsfile

```diff
@@ -69,6 +73,35 @@ for(i = 0; i < buildTypes.size(); i++) {
 builds.failFast = failFast
 parallel builds
 
+stage("Acceptance Tests") {
+    node('docker && highmem') {
+        timestamps {
+            dir("ath") {
+                git url: "https://github.com/jenkinsci/acceptance-test-harness.git"
+
+                // Build container to run ATH
+                def uid = sh returnStdout: true, script: "id -u | tr -d '\n'"
+                def gid = sh returnStdout: true, script: "id -g | tr -d '\n'"
+                def buildArgs = "--build-arg=uid=${uid} --build-arg=gid=${gid} src/main/resources/ath-container"
+                def containerArgs = '-v /var/run/docker.sock:/var/run/docker.sock'
+                def image = docker.build('jenkins/ath', buildArgs)
+
+                unstash "jenkins.war"
+                // Run ATH inside container
+                image.inside(containerArgs) {
```

Comment:
@jglick @abayer @svanoort  IIRC this approach is no longer recommended. amirite?

Reply:
It is safer to `docker run` what you need, and `docker cp` out any artifacts you want the agent JVM to be able to access.

Especially for something like this that we can predict is going to be super flaky and therefore all sorts of people are going to need to know how to reproduce the problem locally. Thus the whole Pipeline script block should be:

* check out some repo
* run a designated script
* run `junit` & `archiveArtifacts`

Reply:
Sorry guys, I do not follow.

Reply:
https://github.com/jenkinsci/docker-workflow-plugin/pull/105 for example.